zio-http/src/main/scala/zio/http/codec/HttpCodec.scala

```diff
@@ -591,16 +591,37 @@ object HttpCodec extends ContentCodecs with HeaderCodecs with MethodCodecs with
 
     def index(index: Int): ContentStream[A] = copy(index = index)
   }
-  private[http] final case class Query[A](name: String, textCodec: TextCodec[A], index: Int = 0)
-      extends Atom[HttpCodecType.Query, A]  {
-    self =>
+
+  private[http] sealed case class Query[A](name: String, textCodec: TextCodec[_], index: Int = 0)
+      extends Atom[HttpCodecType.Query, A] { self =>
```

Comment:
I don't think you should add `QuerySingleValue` or `QueryMultipleValue`. Rather, I would just modify `Query` as follows:

```suggestion
      extends Atom[HttpCodecType.Query, Chunk[A]] { self =>
```

This way the base case produces zero or more values. Then you can derive all the others from this one.

Reply:
Unfortunately, with this approach I couldn't find a way to preserve single value codecs. All `QueryCodecs` will start to produce `Chunk[A]` as a result value, it will break compatibility as well. 

> Then you can derive all the others from this one.

Could you please explain a little bit more your vision on how we can resolve single value codecs type wise?

Reply:
Looked at `HttpCodec` once more and now I see that it has `transform` method which could do exactly what I need.

Reply:
Yes, no need to break backward compatibility, the API can remain exactly the same, because `transform` lets you go from `Chunk[A]` to `A`, and then fail if there is zero (or more than one).